gem/refactorise.py

```diff
@@ -204,18 +202,22 @@ def find_optimal_atomics(self, sum_indices):
         if ilp_prob.status != 1:
             raise AssertionError("Something bad happened during ILP")
 
-        optimal_atomics = [atomic for atomic, _index in iteritems(atomic_index) if ilp_var[_index].value() == 1]
-        other_atomics = [atomic for atomic, _index in iteritems(atomic_index) if ilp_var[_index].value() == 0]
-        optimal_atomics = sorted(optimal_atomics, key=lambda x: self.argument_indices_extent(x), reverse=True)
-        other_atomics = sorted(other_atomics, key=lambda x: self.argument_indices_extent(x), reverse=True)
+        from six.moves import filter, filterfalse
```

Comment:
I imported those in my code snippet to indicate where they came from.  They should probably be imported at the top of the file.